test/integration/rules/color-contrast/text-shadows.html

```diff
@@ -0,0 +1,551 @@
+	<div class="r7">
+		<i>b8-r2-i1</i> <i>b8-r2-i2</i> <i>b8-r2-i3</i> <i>b8-r2-i4</i>
+	</div>
+	<div class="r8">
+		<i>b8-r3-i1</i> <i>b8-r3-i2</i> <i>b8-r3-i3</i> <i>b8-r3-i4</i>
+	</div>
+	<div class="r9">
+		<i>b8-r4-i1</i> <i>b8-r4-i2</i> <i>b8-r4-i3</i> <i>b8-r4-i4</i>
+	</div>
+	<div class="r10">
+		<i>b8-r5-i1</i> <i>b8-r5-i2</i> <i>b8-r5-i3</i> <i>b8-r5-i4</i>
+	</div>
+</section>
+
+<section class="s2 r2" id="b9">
+	<div class="f1">
+		<i>b9-f1-i1</i> <i>b9-f1-i2</i> <i>b9-f1-i3</i> <i>b9-f1-i4</i>
+	</div>
+	<div class="f2">
+		<i>b9-f2-i1</i> <i>b9-f2-i2</i> <i>b9-f2-i3</i> <i>b9-f2-i4</i>
+	</div>
+	<div class="f3">
+		<i>b9-f3-i1</i> <i>b9-f3-i2</i> <i>b9-f3-i3</i> <i>b9-f3-i4</i>
+	</div>
+	<div class="f4">
+		<i>b9-f4-i1</i> <i>b9-f4-i2</i> <i>b9-f4-i3</i> <i>b9-f4-i4</i>
+	</div>
+	<div class="f5">
+		<i>b9-f5-i1</i> <i>b9-f5-i2</i> <i>b9-f5-i3</i> <i>b9-f5-i4</i>
+	</div>
+</section>
+
+<section class="s1 r1" id="b10">
+	<div class="f1">
+		<i>b10-f1-i1</i> <i>b10-f1-i2</i> <i>b10-f1-i3</i> <i>b10-f1-i4</i>
+	</div>
+	<div class="f2">
+		<i>b10-f2-i1</i> <i>b10-f2-i2</i> <i>b10-f2-i3</i> <i>b10-f2-i4</i>
+	</div>
+	<div class="f3">
+		<i>b10-f3-i1</i> <i>b10-f3-i2</i> <i>b10-f3-i3</i> <i>b10-f3-i4</i>
+	</div>
+	<div class="f4">
+		<i>b10-f4-i1</i> <i>b10-f4-i2</i> <i>b10-f4-i3</i> <i>b10-f4-i4</i>
+	</div>
+	<div class="f5">
+		<i>b10-f5-i1</i> <i>b10-f5-i2</i> <i>b10-f5-i3</i> <i>b10-f5-i4</i>
+	</div>
+</section>
+
+<section class="s4 r7" id="b11">
+	<div class="f1">
+		<i>b11-f1-i1</i> <i>b11-f1-i2</i> <i>b11-f1-i3</i> <i>b11-f1-i4</i>
+	</div>
+	<div class="f2">
+		<i>b11-f2-i1</i> <i>b11-f2-i2</i> <i>b11-f2-i3</i> <i>b11-f2-i4</i>
+	</div>
+	<div class="f3">
+		<i>b11-f3-i1</i> <i>b11-f3-i2</i> <i>b11-f3-i3</i> <i>b11-f3-i4</i>
+	</div>
+	<div class="f4">
+		<i>b11-f4-i1</i> <i>b11-f4-i2</i> <i>b11-f4-i3</i> <i>b11-f4-i4</i>
+	</div>
+	<div class="f5">
+		<i>b11-f5-i1</i> <i>b11-f5-i2</i> <i>b11-f5-i3</i> <i>b11-f5-i4</i>
+	</div>
+</section>
+
+<section class="s3" id="b12">
+	<div class="l1">
+		<i>b12-l1-i1</i> <i>b12-l1-i2</i> <i>b12-l1-i3</i> <i>b12-l1-i4</i>
+	</div>
+	<div class="l2">
+		<i>b12-l2-i1</i> <i>b12-l2-i2</i> <i>b12-l2-i3</i> <i>b12-l2-i4</i>
+	</div>
+	<div class="l3">
+		<i>b12-l3-i1</i> <i>b12-l3-i2</i> <i>b12-l3-i3</i> <i>b12-l3-i4</i>
+	</div>
+	<div class="l4">
+		<i>b12-l4-i1</i> <i>b12-l4-i2</i> <i>b12-l4-i3</i> <i>b12-l4-i4</i>
+	</div>
+	<div class="l5">
+		<i>b12-l5-i1</i> <i>b12-l5-i2</i> <i>b12-l5-i3</i> <i>b12-l5-i4</i>
+	</div>
+</section>
+
+<script src="/axe.js"></script>
+<script>
+	// Script added for testing purposes, does not run in automated tests
+	if (window.top == window.self) {
+		axe.run({ runOnly: ['color-contrast'], elementRef: true }, function(
+			err,
+			res
+		) {
+			console.log(res);
+			/*
+      False positives:
+      - b5-r3-i1, b5-r3-i2
+      - b7-r3-i1, b7-r3-i2
+
+      False negatives:
+      - b3-r1-i1
+      - b4-r1-i1, b4-r2-i1
+      - b8-r1-i1, b8-r5-i1
```

Comment:
Left this here just to have a record of it. All these test cases are on the edge of pass and fail. If we ever revisit them, these are the ones I think could be improved on, but I suspect I've put enough time into this it'll never come up.

Reply:
Can you put this comment in the code? Otherwise in a month or two we won't remember what this comment means.